crates/vsock/src/vsock_conn.rs

```diff
@@ -310,9 +318,10 @@ impl<S: AsRawFd + Read + Write> VsockConnection<S> {
             self.fwd_cnt += Wrapping(written_count as u32);
 
             // We send a credit update only when the space available seen
-            // by the transmitter is less than CREDIT_UPDATE_THRESHOLD.
-            let free_space = Wrapping(CONN_TX_BUF_SIZE) - (self.fwd_cnt - self.last_fwd_cnt);
-            if free_space.0 < CREDIT_UPDATE_THRESHOLD {
+            // by the transmitter is less than credit_update_threshold.
+            let free_space =
+                Wrapping(self.tx_buf.get_buf_size()) - (self.fwd_cnt - self.last_fwd_cnt);
```

Comment:
Since we are requesting `self.tx_buf.get_buf_size()` here, perhaps we can avoid introducing `credit_update_threashold` and calculating it each time.